### ISSUE TITLE
Cluster discovery using ECS/K8s

### ISSUE THREAD

**Issue by MichelJansen91** [vernemq/vernemq#1467]
### Environment

- VerneMQ Version: docker-vernemq:latest
- OS: Docker
- Erlang/OTP version (if building from source): N/A
- VerneMQ configuration (vernemq.conf) or the changes from the default:
- Cluster size/standalone: Auto-scaling cluster with at least 2 nodes

### Desired behavior

I am trying to set up an auto-scaling VerneMQ cluster with AWS ECS. I have created a single Task Definition (Similar to a K8s Pod) for setup of single VerneMQ Docker container with environment variables for right configuration).
I am trying to launch multiple replicas /copies of this Task and have them cluster up automatically.
However, I am having problems because the IP address of the Task container is not known beforehand and not fixed either. In order to overcome that problem I tried to use different ways of service discovery, but it seems to me that service discovery is more used for service-to-service communication instead of container-to-container-in-same-service communication. 
I tried to figure out how this is described in the Kubernetes documentation of VerneMQ but I couldn't figure out how it its done. Sidenote: I prefer not to use K8s, but I see it as backup option in case ECS would not work out. 

Basically the issue is similar to #51 and #150, but the posts are from ~4/5 years ago.
I would be curious to see what is the current status, and to know what I am currently missing.

**Comment by nrvmodi** [comment]
Hello @MichelJansen91, Have you tried using rout53 service discovery in aws as per @dergraf  suggested in this [here](https://github.com/vernemq/vernemq/issues/51#issuecomment-148091402)

What issues did you face in that?

**Comment by MichelJansen91** [comment]
Thank you for your reply.
Yes I have tried using the route53 service discovery, but the services were unable to find each other if I tried to fill in the local DNS address as DOCKER_VERNEMQ_DISCOVERY_NODE. I still have to investigate it a little more ( something seemed a bit off with security policies), but also I kind of realized that Kubernetes is quite inevitable for our current setup.
I will try to find some time to work on this

**Comment by MichelJansen91** [comment]
I am currently using the Kubernetes Helm Chart, and that one just works off the shelf. My feeling is that it could also be done with ECS Service as well, but will require a lot of manual configuration compared to the Helm chart. Since I am using Kubernetes now, I will not investigate the ECS issue any further so I am closing the issue.
